app/apollo/models/cluster.schema.js

```diff
@@ -111,4 +113,24 @@ const ClusterSchema = new mongoose.Schema({
 
 ClusterSchema.index({ org_id: 1, cluster_id: 1 }, { unique: true });
 
+// Used to get cluster details for ServiceSubscriptions
+ClusterSchema.statics.getClustersByIds = async function(ids){
+  if(!ids || ids.length < 1){
+    return {};
+  }
+  const clusters = await this.find({ cluster_id: { $in: ids } }, { }, { lean: 1 });
+  const mappedClusters = clusters.map((cluster)=>{
+    //return a BasicCluster
+    return {
+      id: cluster._id,
+      orgId: cluster.org_id,
+      clusterId: cluster.cluster_id,
+      name: cluster.name,
+      registration: cluster.registration,
+    };
```

Comment:
Is there a reason to filter to basic clusters? Maybe for security?
The typical graphql pattern sends back the full object, and the user must explicitly request every attribute they want. So filtering like this doesnt make things easier (like they usually would in rest)

Reply:
Yes, security was my intent.  The 'full' Cluster type includes `resources`, which are customer data that doesn't need to be exposed when querying serviceSubscriptions.  Other Cluster type attributes such as `status`, and `groups` are not necessary for the serviceSubscriptions query, so also excluded.